apps/mobile/src/screens/NftDetailScreen/UniversalNftDetailScreenInner.tsx

```diff
@@ -0,0 +1,248 @@
+import { RouteProp, useNavigation, useRoute } from '@react-navigation/native';
+import { useCallback, useState } from 'react';
+import { ScrollView, StyleSheet, View } from 'react-native';
+import FastImage from 'react-native-fast-image';
+import { useLazyLoadQuery } from 'react-relay';
+import { graphql } from 'relay-runtime';
+import { ShareIcon } from 'src/icons/ShareIcon';
+import { shareUniversalToken } from 'src/utils/shareToken';
+
+import { BackButton } from '~/components/BackButton';
+import { GalleryTouchableOpacity } from '~/components/GalleryTouchableOpacity';
+import { IconContainer } from '~/components/IconContainer';
+import { NftPreviewErrorFallback } from '~/components/NftPreview/NftPreviewErrorFallback';
+import { Pill } from '~/components/Pill';
+import { UniversalNftDetailScreenInnerQuery } from '~/generated/UniversalNftDetailScreenInnerQuery.graphql';
+import { MainTabStackNavigatorParamList, MainTabStackNavigatorProp } from '~/navigation/types';
+import { NftDetailAssetCacheSwapper } from '~/screens/NftDetailScreen/NftDetailAsset/NftDetailAssetCacheSwapper';
+import { useTrack } from '~/shared/contexts/AnalyticsContext';
+import { ReportingErrorBoundary } from '~/shared/errors/ReportingErrorBoundary';
+
+import { InteractiveLink } from '../../components/InteractiveLink';
+import { Markdown } from '../../components/Markdown';
+import { Typography } from '../../components/Typography';
+import { PoapIcon } from '../../icons/PoapIcon';
+import { NftAdditionalDetails } from './NftAdditionalDetails';
+import { NftDetailAsset } from './NftDetailAsset/NftDetailAsset';
+
+const markdownStyles = StyleSheet.create({
+  body: {
+    fontSize: 14,
+  },
+});
+
+const ENABLED_CREATOR = false;
+
+// TODO: Refactor this to merge with existing NftDetailScreen
+
+export function UniversalNftDetailScreenInner() {
+  const route = useRoute<RouteProp<MainTabStackNavigatorParamList, 'NftDetail'>>();
+
+  const query = useLazyLoadQuery<UniversalNftDetailScreenInnerQuery>(
+    graphql`
+      query UniversalNftDetailScreenInnerQuery($tokenId: DBID!) {
+        tokenById(id: $tokenId) {
+          ... on Token {
+            __typename
+            name
+            chain
+            tokenId
+            description
+
+            contract {
+              name
+              badgeURL
+              contractAddress {
+                address
+                chain
+              }
+            }
+            owner {
+              username
+            }
+
+            ...NftAdditionalDetailsFragment
+            ...NftDetailAssetFragment
+            ...shareTokenUniversalFragment
+          }
+        }
+      }
+    `,
+    {
+      tokenId: route.params.tokenId,
+    }
+    // Use one of these if you want to test with a specific NFT
```

Comment:
probably don't need to include this list again since we have it in `NftDetailScreenInner.tsx`

Reply:
Removed it!